server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponseTests.java

```diff
@@ -8,107 +8,238 @@
-            Map<String, IndexFieldCapabilities> fieldCaps = randomFieldCaps.get();
-            for (String index : indices) {
-                inList.add(new FieldCapabilitiesIndexResponse(index, hashing, fieldCaps, true));
+        int numGroups = between(0, 10);
+        for (int g = 0; g < numGroups; g++) {
+            String mappingHash = "mapping_hash_" + g;
+            String group = "group_" + g;
+            List<String> indices = IntStream.range(0, between(1, 10)).mapToObj(n -> group + "_index_" + n).toList();
+            mappingHashToIndices.put(mappingHash, indices);
+        }
+        return mappingHashToIndices;
+    }
+
+    private List<FieldCapabilitiesIndexResponse> randomIndexResponsesWithMappingHash(Map<String, List<String>> mappingHashToIndices) {
+        final List<FieldCapabilitiesIndexResponse> responses = new ArrayList<>();
+        for (Map.Entry<String, List<String>> e : mappingHashToIndices.entrySet()) {
+            Map<String, IndexFieldCapabilities> fieldCaps = randomFieldCaps();
+            String mappingHash = e.getKey();
+            for (String index : e.getValue()) {
+                responses.add(new FieldCapabilitiesIndexResponse(index, mappingHash, fieldCaps, true));
             }
         }
-        int numUngroups = randomIntBetween(0, 5);
-        for (int i = 0; i < numUngroups; i++) {
-            String index = "ungrouped_" + i;
-            final String hashing;
-            final boolean canMatch;
-            Map<String, IndexFieldCapabilities> fieldCaps = Map.of();
-            if (randomBoolean()) {
-                canMatch = false;
-                hashing = UUIDs.randomBase64UUID();
-            } else {
-                canMatch = randomBoolean();
-                hashing = null;
-                if (canMatch) {
-                    fieldCaps = randomFieldCaps.get();
+        return responses;
+    }
+
+    private List<FieldCapabilitiesIndexResponse> randomIndexResponsesWithoutMappingHash() {
+        final List<FieldCapabilitiesIndexResponse> responses = new ArrayList<>();
+        int numIndices = between(0, 10);
+        for (int i = 0; i < numIndices; i++) {
+            String index = "index_without_mapping_hash_" + i;
+            responses.add(new FieldCapabilitiesIndexResponse(index, null, randomFieldCaps(), randomBoolean()));
+        }
+        return responses;
+    }
+
+    public void testWriteResponsesBetweenNewNodes() throws Exception {
+        Map<String, List<String>> mappingHashToIndices = randomMappingHashToIndices();
+        List<FieldCapabilitiesIndexResponse> responseList = CollectionUtils.concatLists(
+            randomIndexResponsesWithMappingHash(mappingHashToIndices),
+            randomIndexResponsesWithoutMappingHash()
+        );
+        assertSerializeOnNewNodes(responseList, mappingHashToIndices);
+        final Version minCompactVersion = Version.CURRENT.minimumCompatibilityVersion();
+        if (minCompactVersion.before(Version.V_8_2_0)) {
```

Comment:
This will be true for all 8.x series, and no longer true for 9.x? But should the test still be around on 9.x when all the supported versions support the index mapping hash? This applied to all places where you used min compat version.

Reply:
We must remove them on 9.x as they will fail.

Reply:
Isn't it enough that they fail on 9.x, and will be handled/removed at that point?